x-pack/filebeat/input/awss3/collector.go

```diff
@@ -245,13 +245,30 @@ func getRegionFromQueueURL(queueURL string, endpoint string) (string, error) {
 // handle message
 func (c *s3Collector) handleSQSMessage(m sqs.Message) ([]s3Info, error) {
 	var msg sqsMessage
-	err := json.Unmarshal([]byte(*m.Body), &msg)
-	if err != nil {
-		c.logger.Debug("sqs message body = ", *m.Body)
-		if jsonError, ok := err.(*json.SyntaxError); ok {
-			return nil, fmt.Errorf("json unmarshal sqs message body failed at offset %d with syntax error: %w", jsonError.Offset, err)
-		} else {
-			return nil, fmt.Errorf("json unmarshal sqs message body failed: %w", err)
+	var bodyJSON struct {
+		TopicArn          string
+		Message           string
+		MessageAttributes map[string]struct{ Value string }
+	}
+	if err := json.Unmarshal([]byte(*m.Body), &bodyJSON); err == nil && bodyJSON.TopicArn != "" {
+		err := json.Unmarshal([]byte(bodyJSON.Message), &msg)
+		if err != nil {
```

Comment:
Seems like these two error checks have duplicate code, could you simplify this part please?

Reply:
Deduplicated this. You are right...